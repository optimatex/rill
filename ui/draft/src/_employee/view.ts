export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
