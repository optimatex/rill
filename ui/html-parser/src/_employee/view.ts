export { React, RC } from '@rill/view';
