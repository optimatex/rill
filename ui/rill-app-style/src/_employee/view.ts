export { createGlobalStyle, css } from '@rill/view';
