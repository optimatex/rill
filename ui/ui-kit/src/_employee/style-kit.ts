export {
  // appLayout
  appLayoutCss,
  screenCss,
  sidebarColumnCss,
  flatColumnCss,
  mainColumnCss,
  // layout
  listCss,
  // Brand
  logCss,
  dualLogoCss,
  // Controller
  trivialAppHeaderCss,
  // _menu
  menuWrapCss,
  menuItemCss,
  // Image
  imageCss,
  imagesListCss,
  imagesGridCss,
  imagesLayoutItemCss,
  imageCropperCss,
  // Tags
  tag,
  tagsList,
  // Ui-primal
  // _grid
  gridRowContainerCss,
  gridrillRowContainerCss,
  gridColumnCss,
  // _card
  cardContainerCss,
  postContainerCss,
  postTitleCss,
  panelCss,
  // _buttons
  buttonCss,
  buttonActiveCss,
  buticonCss,
  // _form
  formContainerCss,
  formFieldFlatCss,
  formFieldColumnCss,
  formFieldErrorCss,
  formLabelTextCss,
  inputCss,
  toggleContainerCss,
  toggleContainerActiveCss,
  toggleRoundCss,
  toggleRoundActiveCss,
  // _article
  paragraphCss,
  nameCss,
  richContentWrapCss,
  // _table
  tableContainerCss,
  tableHeadCss,
  tableRowCss,
  tableColumnCss,
  // Unknown
  // _object
  stringValueCss,
  numberValueCss,
  booleanValueCss,
  emptyValueCss,
  arrayWrapCss,
  arrayItemCss,
  objectKeyCss,
  objectWrapCss,
  objectPairCss,
  // Unsorted
  lightIndicatorCss,
  popoverCss,
  pipeCss,
  hintableLogCss,
  fixedPanelCss,
} from '@rill/style-kit';
