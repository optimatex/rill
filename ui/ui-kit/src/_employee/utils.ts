export {
  toPairs,
  find,
  has,
  path,
  prop,
  curry,
  assoc,
  lensProp,
  view,
  set,
} from '@rill/utils';
