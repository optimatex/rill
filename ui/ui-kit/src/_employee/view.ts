export {
  React,
  useState,
  useCallback,
  // styled-components
  styled,
  // Lens
  // getTestId,
  // Types
  FC,
  RC,
  ReactNode,
  LegacyRef,
  // PropTypes
  IContainerProps,
  // Event types
  FormEvent,
  MouseEvent,
} from '@rill/view';
