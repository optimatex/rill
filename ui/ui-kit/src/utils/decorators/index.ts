export * from './defaultProps';
export * from './testIdOperations';
export * from './testId';
