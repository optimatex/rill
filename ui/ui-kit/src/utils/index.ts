export * from './menu';
export * from './decorators';
