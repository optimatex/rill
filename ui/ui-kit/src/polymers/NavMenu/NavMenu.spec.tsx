import { React } from '_employee/view';
import { render, mockNavMenuScheme_1, mockGetMenuItemProps } from 'test';
import { navMenuId } from 'utils';
import { NavMenu } from '.';

describe('NavMenu', () => {
  const props = {
    scheme: mockNavMenuScheme_1,
    getMenuItemProps: mockGetMenuItemProps,
    testId: navMenuId,
  };

  it('should render without errors', () => {
    const { getAllByRole } = render(
      <NavMenu {...props}>Some content </NavMenu>,
    );
    const el = getAllByRole('link');

    expect(el).toMatchSnapshot();
  });
});
