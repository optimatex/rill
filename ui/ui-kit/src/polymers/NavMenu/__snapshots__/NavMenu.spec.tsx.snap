// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu should render without errors 1`] = `
Array [
  .c1::before {
  font-size: 22px;
  position: relative;
  color: #6b62b1;
  top: px;
  bottom: 1.5px;
  left: px;
  right: 1.5px;
}

.c0 {
  box-sizing: border-box;
  padding: 4px 8px;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 > i {
  margin-right: 12px;
}

<a
    class="c0"
    href="play"
  >
    <i
      class="c1 icon-rewind"
      color="#6b62b1"
      name="rewind"
      size="22"
    />
    Play track
  </a>,
  .c1::before {
  font-size: 15px;
  position: relative;
  color: green;
  top: px;
  bottom: px;
  left: px;
  right: px;
}

.c0 {
  box-sizing: border-box;
  padding: 4px 8px;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 > i {
  margin-right: 12px;
}

<a
    class="c0"
    href="pause"
  >
    <i
      class="c1 icon-menu"
      color="green"
      name="menu"
      size="15"
    />
    Pause track
  </a>,
  .c1::before {
  font-size: 43px;
  position: relative;
  color: pink;
  top: px;
  bottom: px;
  left: px;
  right: px;
}

.c0 {
  box-sizing: border-box;
  padding: 4px 8px;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 > i {
  margin-right: 12px;
}

<a
    class="c0"
    href="next"
  >
    <i
      class="c1 icon-play"
      color="pink"
      name="play"
      size="43"
    />
    Next track
  </a>,
]
`;
